main.py

```diff
@@ -101,9 +106,178 @@ def workflow_1(playlist_length):
+            return False
+        print('What was that?')
+
+
+def find_seed_songs_names():
+    song_names = []
+
+    finding_songs = True
+    while finding_songs:
+        print('Find songs containing the keyword: ', end='')
+        keyword = input().strip().lower()
+
+        song_names = find_songs_by_keyword(keyword)
+
+        if song_names:
+            finding_songs = False
+        else:
+            print(f'Sorry. No songs matched the keyword "{keyword}".')
+            finding_songs = input_bool('Try another keyword?')
+
+    return song_names
+
+
+def select_seed_song_name(song_names):
+    song_name = None
+
+    selecting_song = True
+    while selecting_song:
+        # print('Select the index of the song that you wish to add: ')
+        (index, valid_index) = select_option(song_names,
+            option_label_fn=get_song_option_label, message='Select the index of the song that you wish to add: ')
+
+        if not valid_index:
+            print(f'Sorry. There is no song at index "{index}".')
+            selecting_song = input_bool("Try another index?")
+
+        else:
+            song_name = song_names[index - 1]
+            selecting_song = False
+
+    return song_name
+
+
+def display_seeds(seeds):
+    print(f'Seeds:')
+    for i, song_name in enumerate(seeds, start=1):
+        song = find_spotify_info(song_name)
+        name = song['name']
+        artists = song.artists
+        print(f'[{i}] {name} by {artists}')
+    print()
+
+
+def add_seed(seeds):
+    unique_seeds = set(seeds)
+
+    running = True
+    while running:
+        # Find all songs matching a keyword
+        song_names = find_seed_songs_names()
+        if not song_names:
+            break
+
+        # Select a song from the search results
+        song_name = select_seed_song_name(song_names)
+        if song_name is not None:
+            unique_seeds.add(song_name)
+            print(f'Added: {song_name}')
+
+        running = input_bool('Add another song?:')
+
+    return sorted(unique_seeds)
+
+
+def remove_seed(seeds):
+    updated_seeds = seeds
+    running = True
+
+    while running:
+        print('Select the index of the song that you wish to remove: ')
+        (index, valid_index) = select_option(seeds, option_label_fn=get_song_option_label)
+
+        if not valid_index:
+            print(f'Sorry. There is no song at index "{index}".')
+            running = input_bool("Try another index?")
+
+        else:
+            song_name = updated_seeds.pop(index - 1)
+            print(f'Removed: {song_name}')
+
+            if not updated_seeds:
+                print('No more songs to remove.')
+                running = False
+            else:
+                running = input_bool('Remove another song?')
+
+    return updated_seeds
+
+
 def workflow_2(playlist_length):
-    # TODO: Implementation of WF#2
-    return []
+    seeds = []
+    playlist_songs = []
+    finished_workflow = False
+
+    while not finished_workflow:
```

Comment:
Could we do validation on the seed length as well? Since I'm assuming the seed songs will be part of the playlist, it shouldn't be longer than `playlist_length`? Maybe:
```suggestion
    while not finished_workflow and len(seeds) < playlist_length - 1:
```

Reply:
Okie. I've added this in the latest commits!